v0.15 release notes draft (#3283)
dashpay/dash: doc/release-notes.md

Version 0.14 introduced new LLMQ-based InstantSend system which is designed to be much more scalable
than the legacy one without sacrificing security and which allowed all transactions to be treated as
InstantSend transactions. The legacy system was disabled together with succesful ChainLocks deployment
but we had to support it for a while to ensure smooth transition period. This version finally drops
the legacy system completely.

Read more about ChainLocks: https://github.com/dashpay/dips/blob/master/dip-0008.md
Read more about LLMQ-based InstantSend: https://github.com/dashpay/dips/blob/master/dip-0010.md

Sporks
------
Security level of ChainLocks and LLMQ-based InstantSend made sporks `SPORK_5_INSTANTSEND_MAX_VALUE` and
`SPORK_12_RECONSIDER_BLOCKS` obsolete, so they are removed now. Sporks `SPORK_15_DETERMINISTIC_MNS_ENABLED`,
`SPORK_16_INSTANTSEND_AUTOLOCKS` and `SPORK_20_INSTANTSEND_LLMQ_BASED` have no code logic behind them anymore
because they were used as parts of DIP0003, DIP0008 and DIP0010 activation process which is finished now.
They are still kept and relayed only to ensure smooth operation of v0.14 clients and will be removed in some

> future verions.

Reviewer: ```suggestion
future versions.
```

should help new nodes to quickly catchup on start and detect any potential double-spends as soon as possible.
This should also help wallets to slightly improve UX by showing correct status of unconfirmed transactions
locked via InstandSend, if they were sent while the receiving wallet was offline. Note that bloom-filters
still apply to such `inv` messages, just like the do for transactions and locks that are relayed on a
regular basis.

PrivateSend improvements
------------------------
This version decouples so called Lite Mode and client-side PrivateSend mixing which allows client-side mixing
on pruned nodes running with `--litemode` option. Such nodes will have to also specify the newly redefined
`--enableprivatesend` option. Non-prunned nodes do not have to do this but they can use `--enableprivatesend`
option to disable mixing completely instead. Please note, that specifying this option only does not start mixing
automatically anymore, to make it work that way you should also use the new `--privatesendautostart` option.
You can always control PrivateSend manually though via `privatesend` RPC.

> Thanks to LLMQ-based InstantSend and its ability lock chains of uncinfirmed transactions and not only a single

Reviewer: ```suggestion
Thanks to LLMQ-based InstantSend and its ability lock chains of unconfirmed transactions and not only a single
```

one like in the legacy system PrivateSend mixing speed has improved significantly. In such an environment

> Liquidity Provider Mode which was introduced long time ago to support mixing volume is no longer needed and

Reviewer: ```suggestion
Liquidity Provider Mode which was introduced a long time ago to support mixing volume is no longer needed and
```

> it is removed now. As such `--liquidityprovider` option is not available anymore.

Reviewer: ```suggestion
it is removed now. As such the `--liquidityprovider` option is not available anymore.

Some other improvements were also introduced to speed up mixing e.g. by joining more queues or dropping potential
malicious mixing participants faster by checking some rules earlier etc. Lots of related code was refactored to
further improve its readability which should hopefully make it easier to someone to re-implement PrivateSend
correctly in other wallets if there is a desire to do so.

Wallet changes
--------------
Wallet internals was optimized to significantly improve performance which should be especially notable for huge

> wallets with tens thousands of transactions or more. GUI for such wallets should be much more responsive too now.

Reviewer: ```suggestion
wallets with tens of thousands of transactions or more. GUI for such wallets should be much more responsive too now.
```

> Runnning Masternodes from local wallets was deprecated long time ago and starting from this version we disable

Reviewer: ```suggestion
Running Masternodes from local wallets was deprecated a long time ago and starting from this version we disable
```

wallet functionality on Masternodes completely.

GUI changes
-----------
Qt GUI went through a refresh to follow branding color guides and to make it feel lighter. All old themes besides

> the Traditioinal one (the one with a minimal styling) were removed and instead a new Dark theme was added.

Reviewer: ```suggestion
the Traditional one (the one with a minimal styling) were removed and instead a new Dark theme was added.
```

> In this version we made a lot of optimizatioins to remove varioius lags and lockups, GUI in general should feel

Reviewer: ```suggestion
In this version we made a lot of optimizations to remove various lags and lockups, GUI in general should feel
```

> much more smoother now, especially for huge wallets or when navigating through the masternode list. The later has

Reviewer: ```suggestion
much more smoother now, especially for huge wallets or when navigating through the masternode list. The latter has
```

a few new columns (collateral, owner and voting addresses) which give more options to filter and/or sort the list.
All issues with hi-dpi monitors should also be fixed now.

"Send" popup dialog was slightly tweaked to improve the language and provide a bit more information about inputs
included in the transaction, its size and the actual resulting fee rate. It will also show the number of inputs
a PrivateSend transaction is going to consume and display a warning regarding sender privacy if this number is 10
or higher.

Changes in regtest and devnet p2p/rpc ports
-------------------------------------------

> Default p2p and rpc ports for devnets and regtest were changed to ensure their consistensy and to avoid

Reviewer: ```suggestion
Default p2p and rpc ports for devnets and regtest were changed to ensure their consistency and to avoid
```
